src/senaite/core/browser/worksheet/views/folder.py

```diff
@@ -0,0 +1,476 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of SENAITE.CORE.
+#
+# SENAITE.CORE is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright 2018-2025 by it's authors.
+# Some rights reserved, see README and LICENSE.
+
+import collections
+import json
+
+from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
+from plone.memoize.view import memoize
+from senaite.app.listing import ListingView
+
+from bika.lims import api
+from bika.lims import bikaMessageFactory as _
+from bika.lims.utils import getUsers
+from bika.lims.utils import get_display_list
+from bika.lims.utils import get_link
+from bika.lims.utils import get_progress_bar_html
+from senaite.core.catalog import SETUP_CATALOG
+from senaite.core.catalog import WORKSHEET_CATALOG
+from senaite.core.i18n import translate
+from senaite.core.permissions import AddWorksheet
+from senaite.core.permissions.worksheet import can_add_worksheet
+from senaite.core.permissions.worksheet import can_edit_worksheet
+from senaite.core.permissions.worksheet import can_manage_worksheets
+
+
+class WorksheetsView(ListingView):
```

Comment:
To follow the latest conventions, it's better to rename `folder.py` to `view.py` and move it under the `senaite.core.browser.worksheets` namespace (note the plural worksheet**s**). Then, create a separate module `senaite.core.browser.worksheets.worksheet` to contain all worksheet-specific views and related code.

You can pick the structure of `senaite.core.browsers.clients` as an example.